processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -79,13 +78,42 @@ public ImmutableBitmap getBitmapIndex(BitmapIndexSelector selector)
   @Override
   public ValueMatcher makeMatcher(ValueMatcherFactory factory)
   {
-    throw new UnsupportedOperationException();
+    return factory.makeValueMatcher(
+        dimension, new Predicate<String>()
+        {
+          @Override
+          public boolean apply(String input)
+          {
+            // Assuming that a null/absent/empty dimension are equivalent from the druid perspective
+            return value.equals(Strings.nullToEmpty(fn.apply(Strings.emptyToNull(input))));
+          }
+        }
+    );
   }
 
   @Override
-  public ValueMatcher makeMatcher(ColumnSelectorFactory factory)
+  public ValueMatcher makeMatcher(ColumnSelectorFactory columnSelectorFactory)
   {
-    throw new UnsupportedOperationException();
+    final DimensionSelector dimensionSelector = columnSelectorFactory.makeDimensionSelector(dimension, null);
+    if (dimensionSelector == null) {
+      return new BooleanValueMatcher(value.equals(Strings.nullToEmpty(fn.apply(null))));
+    } else {
+      return new ValueMatcher()
+      {
+        @Override
+        public boolean matches()
+        {
+          final IndexedInts row = dimensionSelector.getRow();
+          final int size = row.size();
+          for (int i = 0; i < size; ++i) {
+            if (value.equals(Strings.nullToEmpty(fn.apply(dimensionSelector.lookupName(row.get(i)))))) {
```

Comment:
This is going to lookup the value in the dictionary on every individual row, meaning that even if your dictionary is 1 element, it will do `n` lookups and conversions.  This can be adjusted to walk the dictionary first, build a set of ids that match and then on each row it can just look those up.

Using a BitSet is probably the best option.